cloud/blockstore/vhost-server/critical_event.cpp

```diff
@@ -0,0 +1,48 @@
+#include "critical_event.h"
+
+#include <cloud/storage/core/libs/diagnostics/logging.h>
+
+#include <util/string/builder.h>
+#include <util/system/spinlock.h>
+
+namespace NCloud::NBlockStore::NVHostServer {
+
+////////////////////////////////////////////////////////////////////////////////
+
+TLog Log;
+TCriticalEvents CriticalEventsAccumulator;
+TAdaptiveLock Guard;
```

Comment:
you can put those global variables inside anonymous namespace to limit visibility